src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -58,19 +58,34 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
 
         internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType, NamedTypeSymbol within, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
-            // We should handle ambiguities once we consider custom modifiers, as we do in overload resolution
-            // https://github.com/dotnet/roslyn/issues/45077
+            MethodSymbol? bestCandidate = null;
+            int bestModifierCountSoFar = -1;
             foreach (var member in containingType.InstanceConstructors)
             {
                 if (HasCopyConstructorSignature(member) &&
                     !member.HasUnsupportedMetadata &&
                     AccessCheck.IsSymbolAccessible(member, within, ref useSiteDiagnostics))
                 {
-                    return member;
+                    // If one has fewer custom modifiers, that is better
+                    // (see OverloadResolution.BetterFunctionMember)
+                    var memberModCount = member.CustomModifierCount();
```

Comment:
>var memberModCount = member.CustomModifierCount(); [](start = 20, length = 50)

Consider doing this only when we detected possible ambiguity, otherwise it is a wasted work.